### ISSUE TITLE
What about tests?

### ISSUE THREAD

**Issue by jaroslaw-weber** [mgattozzi/github-rs#141]
There are maybe two tests which really don't test much. It would be great to be able to write tests for this lib, but how would we go about that?

**Comment by Dylan-DPC** [comment]
HI Jaroslaw

Thanks for showing interest. There are a few more [tests](https://github.com/mgattozzi/github-rs/commit/0eb819e621e6263577bd36d2f222ddfd2f1983a2)  I added last week (not yet released). 

You can follow the same workflow for these integration tests. If you want to write tests, let me know we can discuss this further.

**Comment by mgattozzi** [comment]
Yeah definitely need more tests. Here's some problems that I never really got around to:
1) Token is rate limited. Shouldn't be a problem with a small amount of tests but if you have a lot run quickly it can be a problem
2) How do you mutate something on the site, query it, then revert it?
3) If you do mutate something on the site, the tests run in parallel automatically, how do you get the code to deal with that?

I just never got around to trying to design or solve those problems but there's only so much checking by hand can do.

**Comment by jaroslaw-weber** [comment]
those tests are not integrated with travis unfortunately. we could pass token as mgattozzi said but requests are limited.

the only idea i have is to run mock server before the tests and query the mock test locally. also there would need to be an option for github struct to connect to mock server.

**Comment by Dylan-DPC** [comment]
Ouch yes. Will fix that soon. Thanks

**Comment by Dylan-DPC** [comment]
@mgattozzi is there any way we can get the key (if poss from the encrypted file) for the travis tests?

**Comment by jaroslaw-weber** [comment]
running tests on each PR will exhaust token quickly, we would need to be careful about number of tests so it would be only temporary solution. better to create a simulation (dedicated mock plugin) for exhaustive testing.

**Comment by Dylan-DPC** [comment]
Yes it will. This is just a temporary solution which works fine for current amount of tests + PR we receive.

**Comment by mgattozzi** [comment]
@jaroslaw-weber yeah considering the small amount of PRs for now that rate limit won't be bad for now. Eventually we'll want to mock or maybe contact GitHub about a possible restriction limit being removed for a specific token.

@Dylan-DPC I ended up being to busy tonight to look at it but I'll see if I can quickly set it up before my morning stand up at work tomorrow :)
